### PR TITLE
WIP feat(css): css is preprocessed by csscrush

### LINKED ISSUE #10625: adopt a css preprocessor
A pure php processor can be found here http://the-echoplex.net/csscrush

This would make css more readable and easier to maintain... also it would allow use to define some global vars to be used in css so they could be easier to replace in plugin (think of colors or font family).

### PR DESCRIPTION
**Pull request by jdalsem**
adds csscrush (http://the-echoplex.net/csscrush) capabilities to css views

- [ ] write some docs

fixes #10625 

ref #10316

### CODE REVIEW

**Comment by jdalsem** [comment]
attempt to add preprocessing to Elgg

this change does not brake anything (yet),,, it adds preprocessing to every css view.... don't know if that is the best way... can't use simplecache generate hook as it should always be applied (otherwise css is not working without cache)

currently i only removed some vendor specific css rules (that will be added by csscrush).

csscrush also reduces the css file size (by some 10%)

**Comment by hypeJunction** [comment]
I just looked through their API and I think I love it.

File: engine/lib/views.php [hunk 1]
```diff
@@ -1918,6 +1952,8 @@ function elgg_views_boot() {
 
 	elgg_register_ajax_view('languages.js');
 	
+	elgg_register_plugin_hook_handler('view', 'all', '_elgg_views_preprocess_css');
```
Comment (mrclay) [inline engine/lib/views.php]: Processing all views is a big red flag for me. Limited to /cache requests this could be OK.

**Comment by mrclay** [comment]
> can't use simplecache generate hook as it should always be applied (otherwise css is not working without cache)

Can't we add a new hook before it for this? If we define a variable somewhere we want to be able to use it even in smaller component views.

**Comment by mrclay** [comment]
How much time does this add to the current /cache CSS response with simplecache off? I know this avoids needing a build step, but I wonder how much pain this will add during development.

File: composer.json [hunk 2]
```diff
@@ -28,6 +28,7 @@
         "bower-asset/jquery-form": "^3.51",
         "bower-asset/jquery-colorbox": "^1.5.14",
         "bower-asset/sprintf": "~1.0.3",
+        "css-crush/css-crush": "^2.4.0",
```
Comment (mrclay) [inline composer.json]: See if you can add this without updating all our dependencies. Would rather project-wide updates happen in a separate PR.

**Comment by mrclay** [comment]
Apparently `simplecache:generate` is triggered even when simplecache is off, so you just need to register for it.

**Comment by mrclay** [comment]
Replaced by #10669

**Pull request closed without merging**